third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -121,10 +133,12 @@ func resourceDataflowJobCreate(d *schema.ResourceData, meta interface{}) error {
 	params := expandStringMap(d, "parameters")
 
 	env := dataflow.RuntimeEnvironment{
-		TempLocation:        d.Get("temp_gcs_location").(string),
-		Zone:                zone,
 		MaxWorkers:          int64(d.Get("max_workers").(int)),
+		Network:             d.Get("network").(string),
```

Comment:
When storing a reference, we'll generally wrap it in `ConvertSelfLinkToV1` such as in https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/resources/resource_compute_instance.go#L891. This canonicalises non-`v1` self links from compute products to `v1`.

(This is a little redundant with the diff suppress func, but ends up handling a couple additional corner cases when referencing this resource from other resources)